### ISSUE TITLE
Jenkins un-idling not working

### ISSUE THREAD

**Issue by ldimaggi** [openshiftio/openshift.io#3517]
Seeing this behavior:
* Reset user env
* Observe Jenkins and Content Repo pods start up
* Wait until both pods idle
* Create a new app
* Observe Jenkins and Content Repo pods do not un-idle

![screenshot from 2018-05-08 19-23-50](https://user-images.githubusercontent.com/642621/39789351-56a8d0dc-52fd-11e8-9e6f-0edc7636ac9f.png)

![3](https://user-images.githubusercontent.com/642621/39789350-569a539a-52fd-11e8-82fe-0ded9e53648f.png)

**Comment by ldimaggi** [comment]
The only event seen is:
```
Replication Controller Scaled | Scaled replication controller "jenkins-1" from 0 to 1
```

**Comment by joshuawilson** [comment]
related to #3356

**Comment by containscafeine** [comment]
@ldimaggi were the pods still being showed as idled in the UI? This could be an intermittent error of content-repository service not responding to the Jenkins init-container which @piyush1594 and @rupalibehera have faced in the past.

**Comment by containscafeine** [comment]
One way to test if this is the case is by checking if there are continuous retries in the Jenkins' init container even though the content repository is up; run the command `oc logs -c content-repository-init -f <jenkins pod name>`

**Comment by piyush1594** [comment]
I also faced this thing today

Steps:
My Jenkins and content repo were idle.
I created an application
Content repository got unidle but Jenkins pod does not come up and get stuck in init container stage. It's not able to get the response from content-repository for the curl request it is making

```
oc logs -c content-repository-init jenkins-1-gsgnp
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0
curl: (28) Operation timed out after 4001 milliseconds with 0 out of -1 bytes received
waiting for content-repository ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0
curl: (28) Operation timed out after 4001 milliseconds with 0 out of -1 bytes received
waiting for content-repository ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0

```

After that restarting the Jenkins pod works.

**Comment by ldimaggi** [comment]
Yes - the pods were shown as idle in the UI.

**Comment by ljelinkova** [comment]
Is there any progress on this issue? The E2E tests are still failing from time to time because of that.

**Comment by containscafeine** [comment]
@ljelinkova can you share the logs for the failures? I'd also like to see the logs of content-repository-init init container in jenkins deployment in the e2e tests, if those are logged.

**Comment by ljelinkova** [comment]
@containscafeine Unfortunately, the tests are set to reset the whole environment after each run. 

However, if you can specify which logs could help you debug the issue, I might try to store those logs during the test run.

**Comment by ljelinkova** [comment]
I'll add the logs retrieving into our tests and post here if we have any results.

**Comment by piyush1594** [comment]
We belive this to be linked with https://github.com/openshiftio/openshift.io/issues/3299

**Comment by ppitonak** [comment]
@containscafeine my Jenkins (project ppitonak-jenkins) didn't un-idle correctly
1. When I go to https://jenkins.openshift.io/ I see the message that it was idled
2. OpenShift console shows this 
![osio_unidle_jenkins](https://user-images.githubusercontent.com/445100/41350241-c70a1b34-6f12-11e8-98c8-55ffe19ee458.png)
3. `oc logs -c content-repository-init <jenkins pod name>` shows this every couple of seconds:
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:03 --:--:--     0curl: (7) Failed connect to content-repository:80; No route to host
waiting for content-repository ...
```

**Comment by containscafeine** [comment]
@ppitonak great, so the init container is failing to make connection to the content-repository. This confirms the link with https://github.com/openshiftio/openshift.io/issues/3299, so well, maybe we're blocked on that :man_shrugging: 
A workaround is to kill the pod, so the pod's networking stack gets reset (if there's any such thing), and the new pod comes up just fine.
This workaround can be scripted maybe, but that's just another ugly hack.

**Comment by ppitonak** [comment]
@containscafeine which pod am I supposed to kill? Jenkins?

**Comment by containscafeine** [comment]
@ppitonak yes.

**Comment by ppitonak** [comment]
@containscafeine How do I do that? It seems that I don't have permissions to do anything in ppitonak-jenkins project

**Comment by containscafeine** [comment]
@ppitonak that's weird. Can you check with @kbsingh or @aditya-konarde about the permissions?

**Comment by ppitonak** [comment]
For the record, it's expected that users cannot restart Jenkins pod.

**Comment by hrishin** [comment]
@ppitonak please request admin permission from @aslakknutsen

**Comment by piyush1594** [comment]
This is WIP here fabric8-services/fabric8-jenkins-proxy#289

**Comment by jiekang** [comment]
This  issue also affects users in production, not just the E2E tests; has there been any further work? Fortunately in production you do have permissions to scale Jenkins pods up and down, but it is frustrating to need to do this whenever Jenkins is idle.

**Comment by piyush1594** [comment]
Its WIP and can be tracked here https://github.com/fabric8-services/fabric8-jenkins-proxy/issues/289

**Comment by ppitonak** [comment]
There is a big groups of cross-linked related issues.

Right now, team is trying to solve the issue of Jenkins booting too slow (#3956, PR https://github.com/fabric8-services/fabric8-tenant-jenkins/pull/114)

**Comment by ppitonak** [comment]
There is a new version of Jenkins in both prod-preview and prod. Un-idling still gets stuck from time to time.

**Comment by kishansagathiya** [comment]
> There is a new version of Jenkins in both prod-preview and prod. Un-idling still gets stuck from time to time.

@ppitonak It would be because many jenkins are getting stuck at the terminating stage. It is being tracked on housekeeping repo. Apart from that we had a task to remove init container hack, which was causing this and is done.

**Comment by ppitonak** [comment]
@kishansagathiya there is a new Jenkins in prod and AFAIK the issue with stuck terminating pods is solved now but the Jenkins is still not un-idling from time to time.

What is the next step?

**Comment by kishansagathiya** [comment]
@sthaha 
When the issue was opened, it was because init container hack in Jenkins.
At the moment
- Init Container hack is removed
- Content repo is removed and jenkins is given more resources

But as @ppitonak mentions 
> Jenkins is still not un-idling from time to time.

These definitely is because of reasons different than earlier. I say we close this one and track current problems with un-idling on a different issue. We already have a mesh of issues because of this issue.

@ppitonak @sthaha wdyt?

**Comment by ppitonak** [comment]
I would prefer keeping this issue open... you cannot be 100% sure that it was caused only by init container hack. Creating a new issue makes sense if a bug is fixed and later a regression appears. This one was never fixed.

Could you list related **open** issues here? I only know about https://github.com/fabric8-services/fabric8-jenkins-idler/issues/258

**Comment by kishansagathiya** [comment]
@ppitonak 
We should close https://github.com/fabric8-services/fabric8-jenkins-idler/issues/258 as well since it makes sense only if we have init-container. Only in that situation, reseting might help us avoid openshift networking bug.

**Comment by sthaha** [comment]
> I would prefer keeping this issue open... you cannot be 100% sure that it was caused only by init container hack. Creating a new issue makes sense if a bug is fixed and later a regression appears. This one was never fixed.

@ppitonak 

IMO, the earlier bugs must be fixed by now. How often are you able to reproduce the issue by repeating the steps in OP ?

1. Reset user env
1. Observe Jenkins and Content Repo pods start up
1. Wait until both pods idle
1. Create a new app
1. Observe Jenkins and Content Repo pods do not un-idle


Could you please post
1. the state of Jenkins pods
1. logs if jenkins pod even starts
1. the jenkins image tag in use
1. the cluster

**Comment by ppitonak** [comment]
@sthaha ok, OP mentions content repo explicitly, although content repo was not the root cause, I agree to close this issue and create a new one.
